controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -37,6 +37,10 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 ) (ctrl.Result, error) {
 	logger := controlPlane.Logger()
 
+	if kcp.Spec.RolloutStrategy == nil && kcp.Spec.RolloutStrategy.RollingUpdate == nil {
+		return ctrl.Result{}, errors.New("rolloutStrategy is not set")
+	}
```

Comment:
Isn't this checked in a defaulting and validartion webhook?

Reply:
Yes, this was checked in both, so I removed the this.